### ISSUE TITLE
Playback issue

### ISSUE THREAD

**Issue by alexomics** [LooseLab/readfish#345]
Hi @alexomics,

I have same RUN ERROR issue on my demo run with in minutes of loading MinION data using playback option in MinKnow UI. I have a MinKnow with Dorado and running in a UBUNTU 22

Not sure where should I start? I currently struck at this step - https://github.com/LooseLab/readfish?tab=readme-ov-file#configuring-bulk-fast5-file-playback

![Screenshot from 2024-04-09 10-31-19](https://github.com/LooseLab/readfish/assets/8802351/23d83acb-84b8-48bf-85ee-2ef3a5e38907)

Wondering, what are permission error I should be checking.
I do have full permission on 5555

```
$ ll /tmp/.guppy/
srwxrwxrwx 1 minknow minknow 0 Mar 28 09:04 5555
```

Thanks for your help,
sam

_Originally posted by @bioteksampath in https://github.com/LooseLab/readfish/issues/337#issuecomment-2045925085_

**Comment by alexomics** [comment]
@bioteksampath

Hi Sam, 

I’ve moved this to a new issue for now. Which playback file are you using?

**Comment by bioteksampath** [comment]
Hi @alexomics 
Thanks for your prompt response, i'm using the following file from the demo.
`GXB02001_20230509_1250_FAW79338_X3_sequencing_run_NA12878_B1_19382aa5_ef4362cd.fast5`

**Comment by alexomics** [comment]
Okay, that’s the right file. Can you find the error message from the control server log? It should be at:
```
/var/log/minknow/MS00000/control_server_log_0.txt
```

**Comment by bioteksampath** [comment]
i do have mutiple log files but this is the latest one

`(base) sap223@gifs-c36qc14:/home/.gifs/sap223/Desktop/readfish_demo$ cat /var/log/minknow/MS00000/control_server_log-0.txt 
2024-04-09 10:19:57.300601    INFO: starting_up (control)
    hostname: gifs-c36qc14.usask.ca
    system: ubuntu 20.04
            Distribution:           23.11.7 (STABLE)
            MinKNOW Core:           5.8.6
            Bream:                  7.8.2
            Protocol configuration: 5.8.6
            Dorado (build):          0.0.0.19120+441f78764
            Dorado (connected):      7.2.13+fba8e8925
            
2024-04-09 10:19:57.302232 WARNING: partially_raised_file_limits (control_server)
    actual: 1024
    target: 4096
2024-04-09 10:19:57.302321    INFO: auth_guest_mode (rpc)
    value: local_only
2024-04-09 10:19:57.303249    INFO: external_offload_service_not_configured (script)
    auth_token_file: 
    external_service_port: 0
2024-04-09 10:19:57.304133    INFO: flow_cell_position_instantiated (mgmt)
    device_id: MS00000
    hardware_type: MINION_USB
    os_identifier: 
    simulated: true
2024-04-09 10:19:57.304176    INFO: active_device_set (mgmt)
    identifier: 0x7f16a40023b0
2024-04-09 10:19:57.304204    INFO: successfully_read_flow_cell_data (mgmt)
    attempt: 1
    flow_cell_data: <empty>
2024-04-09 10:19:57.306726    INFO: manager_notified_new_auth_token (util)
    expires_in: 7510692439548057758ns
    token_type: internal
2024-04-09 10:19:57.306747    INFO: manager_notified_new_auth_token (util)
    expires_in: 88442360630897ns
    token_type: local
2024-04-09 10:19:57.354926    INFO: device_discovered (engine)
    device_id: MS00000
2024-04-09 10:19:58.349872    INFO: read_eeprom_id (mgmt)
    eeprom_id: 4275878400
2024-04-09 10:19:58.355176    INFO: flowcell_discovered (engine)
    asic_id: 131070
    asic_id_eeprom: 4275878400
    flow_cell_id: 
2024-04-09 10:19:58.805690    INFO: firmware_component (rpc)
    component_name: MinION FPGA
    serial_number: 
    version: 1.0.1
2024-04-09 10:19:58.805765    INFO: firmware_component (rpc)
    component_name: USB
    serial_number: 
    version: 1.0.8
2024-04-09 10:21:43.734719    INFO: create_request_for_disabling_OS_standby (util)
2024-04-09 10:21:43.734899    INFO: sending_telemetry_message (ping)
    data: {"monitor":"protocol_start","protocol_output_dir":"/var/lib/minknow/data/./t2/t2s2/20240409_1021_MS00000_MS00000_c5218bcb","script_arguments":["--f...
2024-04-09 10:21:43.736149    INFO: protocol_started (script)
    flowcell_id: MS00000
    host_serial_number: 
    identity: ['']
    library_id: t2s2
    output_path: /var/lib/minknow/data/./t2/t2s2/20240409_1021_MS00000_MS00000_c5218bcb
    output_reads_dir: /var/lib/minknow/data/.
    position_id: MS00000
    protocol_group_id: t2
    protocol_output_pattern: {protocol_group_id}/{sample_id}/{start_time}_{device_id}_{flow_cell_id}_{short_protocol_run_id}
    run_id: c5218bcb-02b3-43a0-ada8-9ac255709794
    sample_id_to_barcode: []
    script_args: -s conf/package/utility/protocol_selector.py start sequencing/sequencing_MIN106_DNA:FLO-MIN106:SQK-LSK109 --fast5=on --pod5=off --fastq=on --bam=off --generate_bulk_file=off --active_channel_selection=on --base_calling=on --fast5_reads_per_file=4000 --fast5_data vbz_compress --fastq_reads_per_file=4000 --fastq_data compress --mux_scan_period=1.5 --pore_reserve=on --guppy_filename=dna_r9.4.1_450bps_fast.cfg --read_filtering min_qscore=8 --read_splitting enable=off --min_read_length=200 --simulation=/home/sap223/gifs/Desktop/readfish_demo/GXB02001_20230509_1250_FAW79338_X3_sequencing_run_NA12878_B1_19382aa5_ef4362cd.fast5
    script_path: sequencing/sequencing_MIN106_DNA:FLO-MIN106:SQK-LSK109
2024-04-09 10:21:43.736346    INFO: disk_space_info_user (rpc)
    filesystem_id: /
    space_remaining: 163
    unit: GB
2024-04-09 10:21:44.618448    INFO: waiting_for_device_ready_for_acquisition (mgmt)
2024-04-09 10:21:45.854335    INFO: switching_to_basecaller_config (basecalling)
    new_filename: dna_r9.4.1_450bps_fast.cfg
2024-04-09 10:21:45.855110    INFO: basecaller_connection_changed_state (basecalling)
    basecaller_error_string: [no_error] 
    client_name: MS00000_config_preload_203ff53a-6796-48b1-986c-643acaef5f73
    next_state: preparing_connection (1)
    previous_state: disconnected (0)
2024-04-09 10:21:45.855255    INFO: basecaller_connection_changed_state (basecalling)
    basecaller_error_string: [no_error] 
    client_name: MS00000_config_preload_203ff53a-6796-48b1-986c-643acaef5f73
    next_state: loading_config (2)
    previous_state: preparing_connection (1)
2024-04-09 10:21:45.856787    INFO: basecaller_connection_changed_state (basecalling)
    basecaller_error_string: [no_error] 
    client_name: MS00000_config_preload_203ff53a-6796-48b1-986c-643acaef5f73
    next_state: loading_alignment_index (3)
    previous_state: loading_config (2)
2024-04-09 10:21:45.856832    INFO: basecaller_connection_changed_state (basecalling)
    basecaller_error_string: [no_error] 
    client_name: MS00000_config_preload_203ff53a-6796-48b1-986c-643acaef5f73
    next_state: loading_bed_file (4)
    previous_state: loading_alignment_index (3)
2024-04-09 10:21:45.856842    INFO: basecaller_connection_changed_state (basecalling)
    basecaller_error_string: [no_error] 
    client_name: MS00000_config_preload_203ff53a-6796-48b1-986c-643acaef5f73
    next_state: connecting (5)
    previous_state: loading_bed_file (4)
2024-04-09 10:21:45.858650    INFO: basecaller_connection_changed_state (basecalling)
    basecaller_error_string: [no_error] 
    client_name: MS00000_config_preload_203ff53a-6796-48b1-986c-643acaef5f73
    next_state: connected (6)
    previous_state: connecting (5)
2024-04-09 10:21:45.941750    INFO: protocol_phase_control_started (rpc)
2024-04-09 10:21:45.942007    INFO: base_device.wait_for_temperature (user_messages)
    target: 34.0
    timeout: 300
2024-04-09 10:21:45.942325    INFO: protocol_phase_changed (script)
    phase: PHASE_INITIALISING
2024-04-09 10:21:46.855043    INFO: basecaller_client_created (basecalling)
    client_name: MS00000_config_preload_203ff53a-6796-48b1-986c-643acaef5f73
2024-04-09 10:21:46.855078    INFO: successfully_created_basecall_client_with_config (basecalling)
    alignment_options: 
    barcoding_options: 
    basecaller_config_name: dna_r9.4.1_450bps_fast.cfg
    lamp_options: 
2024-04-09 10:21:46.856067    INFO: basecaller_connection_changed_state (basecalling)
    basecaller_error_string: [no_error] 
    client_name: MS00000_config_preload_203ff53a-6796-48b1-986c-643acaef5f73
    next_state: disconnected (0)
    previous_state: connected (6)
2024-04-09 10:22:01.962247    INFO: wait_for_temperature_complete (rpc)
    elapsed_time: 16.0189
    has_secondary_temp_limits: true
    primary_temperature_begin: 35
    primary_temperature_end: 34
    primary_temperature_tolerance: 0.1
    result: ReachedTemperature
    secondary_temperature_begin: 35.0431
    secondary_temperature_end: 34.0329
    target_primary_temperature: 34
    unstable_time: 1.00093
2024-04-09 10:22:03.066579    INFO: sending_telemetry_message (ping)
    data: {"bulk_enabled":false,"fastq_enabled":true,"monitor":"exp_start","multi_fast5_enabled":true}
2024-04-09 10:22:03.098119   ERROR: exception_during_observer_callback (state_control)
    callback_mode: transitioning
    cause: Dynamic exception type: boost::filesystem::filesystem_error
           std::exception::what: boost::filesystem::status: Permission denied [system:13]: "/home/sap223/gifs/Desktop/readfish_demo/GXB02001_20230509_1250_FAW79338_X3_sequencing_run_NA12878_B1_19382aa5_ef4362cd.fast5"
           
    observer: MantaControl
2024-04-09 10:22:03.492513    INFO: data_acquisition_starting (engine)
    acquisition_run_id: 9824c93cc5f3296457bbe0e5608f6b17889f26e1
    options: allow_file_output=true, enable_analysis=true, generate_reports=true, send_basecalling_metrics=true, send_sequencing_read_metrics=true, generate_final_summary=true
2024-04-09 10:22:03.497570   ERROR: failed_to_finish_transition (state_control)
    destination: CreateSharedData
    observer: MantaControl
    operation_begin: 2024-04-09T16:22:03.010772
    operation_end: 2024-04-09T16:23:03.009772
    source: Idle
    stage: transition
2024-04-09 10:22:03.597656    INFO: data_acquisition_finished (engine)
    acquisition_run_id: 9824c93cc5f3296457bbe0e5608f6b17889f26e1
2024-04-09 10:22:03.597688    INFO: stop_processing_unblocks_called (mgmt)
2024-04-09 10:22:03.597713 WARNING: called_stop_processing_unblocks_while_not_processing (mgmt)
2024-04-09 10:22:33.599492 WARNING: forcibly_set_statistic_to_finished (pipeline_stats)
    in_function: <unknown type>
    stat_name: <unknown type>
2024-04-09 10:22:33.599893 WARNING: forcibly_set_statistic_to_finished (pipeline_stats)
    in_function: <unknown type>
    stat_name: <unknown type>
2024-04-09 10:22:33.600049 WARNING: forcibly_set_statistic_to_finished (pipeline_stats)
    in_function: <unknown type>
    stat_name: <unknown type>
2024-04-09 10:22:33.600192 WARNING: forcibly_set_statistic_to_finished (pipeline_stats)
    in_function: <unknown type>
    stat_name: <unknown type>
2024-04-09 10:22:33.600319 WARNING: forcibly_set_statistic_to_finished (pipeline_stats)
    in_function: <unknown type>
    stat_name: <unknown type>
2024-04-09 10:22:33.600465 WARNING: forcibly_set_statistic_to_finished (pipeline_stats)
    in_function: <unknown type>
    stat_name: <unknown type>
2024-04-09 10:22:33.601490    INFO: sending_telemetry_message (ping)
    data: {"monitor":"exp_stop","stop_reason":"INTERNAL_ERROR"}
2024-04-09 10:22:33.623661   ERROR: stopping_protocol_due_to_internal_error (control)
2024-04-09 10:22:33.623984    INFO: killing_protocol (script)
    run_id: c5218bcb-02b3-43a0-ada8-9ac255709794
    script_path: sequencing/sequencing_MIN106_DNA:FLO-MIN106:SQK-LSK109
2024-04-09 10:22:33.632163    INFO: protocol_phase_control_ended (rpc)
2024-04-09 10:22:33.656313    INFO: protocol_finished_internal_error (script)
    run_id: c5218bcb-02b3-43a0-ada8-9ac255709794
2024-04-09 10:22:33.656511    INFO: sending_telemetry_message (ping)
    data: {"acquisitions":[],"monitor":"protocol_end","script_name":"sequencing/sequencing_MIN106_DNA:FLO-MIN106:SQK-LSK109","stop_reason":"FINISHED_WITH_ERR...
2024-04-09 10:22:33.656771    INFO: clear_request_for_disabling_OS_standby (util)
2024-04-09 10:22:34.024178    INFO: rpc_error_occured (rpc)
    details: {
                 "error_message": "Failed to start run: Error in observer 'MantaControl': boost::filesystem::status: Permission denied [system:13]: \"/home/sap223/gifs/Desktop/readfish_demo/GXB02001_20230509_1250_FAW79338_X3_sequencing_run_NA12878_B1_19382aa5_ef4362cd.fast5\"", 
                 "exit_code": -1
             }
    path: /minknow_api.acquisition.AcquisitionService/start
    status: UNKNOWN
    user_identity: ['']
2024-04-09 10:32:14.897521    INFO: manager_notified_hw_state_change (util)
    new_identifier: 
    new_state: unavailable
2024-04-09 10:32:14.897583    INFO: manager_requested_exit (util)
    when: asap
2024-04-09 10:32:14.897826    INFO: active_device_detached (mgmt)
    identifier: 0x7f16a40023b0
2024-04-09 10:32:14.900639    INFO: minion_disconnected (engine)
2024-04-09 10:32:15.658833    INFO: exiting (control_server)`

**Comment by alexomics** [comment]
So the error from that log is that MinKNOW cannot access the bulk file. I would try moving it to MinKNOW’s data folder at 
```
/var/lib/minknow/data
```

make sure to update the simulation location and try again

**Comment by bioteksampath** [comment]
Hi @alexomics
Thanks for your response.
yes, Minknow recognise input.fast5 from minknow/data folder but not from other locations. strange.

however i have couple of other issues, hope you can provide some feedback.
Since i have Dorado basecaller (not guppy) do i need to change anything in the .toml file, because i got below error.

 RuntimeError: The dna_r9.4.1_450bps_fast base-calling config listed in the readfish config TOML is not suitable for this flowcell and kit `combination.`

log file.
[Ap10test.log](https://github.com/LooseLab/readfish/files/14950881/Ap10test.log)

`toml file

# Basecaller configuration
[caller_settings.guppy]
#               ^^^^^^ - ".guppy" specifies our chosen basecaller
# Guppy base-calling configuration file name
config = "dna_r9.4.1_450bps_fast"
# Address of the guppy basecaller - The default address for guppy is ipc:///tmp/.guppy/5555.
address = "ipc:///tmp/.guppy/5555"
# Fastq output for individual reads. This is OPTIONAL - as these files can become quite large.
# Remove line to disable.
debug_log = "live_reads.fq"

# Aligner Configuration
[mapper_settings.mappy]
#               ^^^^^^ - ".mappy" specifies mappy as the aligner. Use mappy_rs for the multithreaded rust version (required on PromethION)
# Alignment reference to use. Should be either FASTA or an MMI
fn_idx_in = "/path/to/hg38.mmi"
# Optional PAF output for live alignments.
# Remove line to disable.
debug_log = "live_alignments.paf"
# Number of threads for indexing (mappy and mappy-rs) and mapping (mappy-rs only)
n_threads = 24

# Region Configuration - see https://looselab.github.io/readfish/toml.html#analysis-regions for more information.
# Definitions of "unblock", "proceed" and "stop_receivings" are as follows

    # proceed: Allow one more chunk to be captured, before trying to make another decision, i.e proceed for now

    # unblock: Unblock the read

    # stop_receiving: Allow the read to be sequenced, and stop receiving signal chunks for it.
# This region will enrich for reads mapping to chr20 and chr21.
[[regions]]
name = "hum_test"
min_chunks = 1 # minimum number of chunks before a decision can be made
max_chunks = 4 # maximum number of chunks to use in decision making - after this perform the above_max_chunks action
targets = ["chr20", "chr21"]  # Genomic targets for this region
single_on = "stop_receiving"  # Action to take if there is one mapping on target.
multi_on = "stop_receiving"   # Action to take if there is more than one mapping, with at least one target.
single_off = "unblock"        # Action to take if there is one mapping and it is off target
multi_off = "unblock"         # Action to take if there are multiple mappings, where all are off target.
no_seq = "proceed"            # Action to take if there is no sequence information
no_map = "proceed"            # Action to take if there is no mapping information
above_max_chunks = "unblock"  # Action to take if the number of chunks received is above max_chunks
below_min_chunks = "proceed"  # Action to take if the number of chunks received is below min_chunks
`

**Comment by alexomics** [comment]
I can see from the control server log that you are using the R10 (5KHz) bulk file for playback. As such the `dna_r9.4.1_450bps_fast` config cannot be used for base calling. I would recommend using the `dna_r10.4.1_400bps_5khz_fast` config instead. 
For the caller settings:
```toml
[caller_settings.guppy]
config = "dna_r10.4.1_e8.2_400bps_5khz_fast"
address = "ipc:///tmp/.guppy/5555"
debug_log = "live_reads.fq"
```

---

I can also see from the TOML file you uploaded that you've set the `n_threads` option for the aligner. To make use of multi-threaded alignment you will need to use the `mappy_rs` option and set the correct `fn_idx_in` path:

```TOML
[mapper_settings.mappy]
fn_idx_in = "/path/to/hg38.mmi"
debug_log = "live_alignments.paf"
n_threads = 24
```

**Comment by bioteksampath** [comment]
Thanks @alexomics 

I was able to run the demo successfully. I have some additional questions regarding my own research, and I'm hoping you could provide some feedback:

1. I want to genotype 500 structural variants (SVs) greater than 5kb across 50 lines. What would be the best approach for this?
2. Concerning target sizes in the target.bed file, I've heard that each target should be around 50kb to ensure good sequence coverage. Could you share your thoughts on the minimum size and maximum number of targets?
3. How many libraries should be maximally pooled or barcoded in a single MinION sequencing run for a genome size of 1Gb?
4. Could you recommend any downstream analysis tools for validation and visualization?

Apologies for the barrage of questions.

**Comment by mattloose** [comment]
HI @bioteksampath ,

The above questions aren't issues with readfish and also are not linked to the original issue (which is now resolved). 

It is helpful to keep these things separate. I shall close this issue and reference your comment in a new discussion - #346

**Comment by bioteksampath** [comment]
Thanks, @mattloose and @alexomics, 

I wanted to share the steps I took to address the issues we encountered with the Nanopore adaptive sequencing setup, I believe it will be useful for others.

1. **Minknow File Access**: I found that Minknow was unable to access the demo fast5 (R10 data) file from a directory other than `/var/lib/minknow/data`. To resolve this, I moved the file to this folder. Please note that writing fast5 files to `/var/lib/minknow/data` requires sudo permission.

2. **Write Permission**: To allow Minknow to write to `/tmp/.guppy/5555`, I used the command `chmod 777 /tmp/.guppy/5555`. However, obtaining write permission may require a request to the sudo user.

3. **Configuration Update**: I made changes to the `.toml` configuration information. It is compatible with Dorado caller. Here are the updated settings:

   For Guppy/Dorado:
   ```
   [caller_settings.guppy]
   config = "dna_r10.4.1_e8.2_400bps_5khz_fast"
   address = "ipc:///tmp/.guppy/5555"
   debug_log = "live_reads.fq"
   ```

   For Multi-Threads:
   ```
   [mapper_settings.mappy_rs]
   fn_idx_in = "/path/to/hg38.mmi"
   debug_log = "live_alignments.paf"
   n_threads = 24
   ```

   For Single Thread:
   ```
   [mapper_settings.mappy]
   fn_idx_in = "/path/to/hg38.mmi"
   debug_log = "live_alignments.paf"
   n_threads = 4
   ```

4. **Additional Information**: If using a targets bed file, ensure it is comma-separated. Here's an example format:

   ```
   targets = ["/path/to/targets.bed"]
   Chr1
   Scf777,510021,510052,+
   Scf4546,108529,108529,+
   Chr11,1,58419410,+
   Chr12,1,5639592,+
   ```

These adjustments should help resolve the issues we encountered. 

Hope this helps someone!
Sam

**Comment by mattloose** [comment]
Could you check your bed file format please? 

The format you provide above is not a bed file format - the bed file format spcs are available here https://github.com/samtools/hts-specs/blob/master/BEDv1.pdf and are explicitly not comma separated - rather they are tab or whitespace separated, which readfish does support.

Could you share the file that you used which did not work?

**Comment by bioteksampath** [comment]
@mattloose thanks for pointing out.|

Strange is i can not validate the .bed format, but target csv 

```
(rf_con) cxo314@gifs-c36qc14:/data/reference/canola$ head targets_315Svs.bed
777	510021	520052	+
4546	108529	109529	+
N11	1	58419410	+
N12	1	5639592	+
N13	1	5639592	+
N19	1	5639592	+
N2	68644	69237	+

```

```
(rf_con) cxo314@gifs-c36qc14:/data/reference/canola$ readfish validate canola_test_1600_N11_barcode1-4_bed.toml 
2024-04-15 12:55:49,862 readfish /home/cxo314/gifs/.conda/envs/rf_con/bin/readfish validate canola_test_1600_N11_barcode1-4_bed.toml
2024-04-15 12:55:49,862 readfish command='validate'
2024-04-15 12:55:49,862 readfish log_file=None
2024-04-15 12:55:49,862 readfish log_format='%(asctime)s %(name)s %(message)s'
2024-04-15 12:55:49,862 readfish log_level='info'
2024-04-15 12:55:49,862 readfish no_check_plugins=False
2024-04-15 12:55:49,862 readfish no_describe=False
2024-04-15 12:55:49,862 readfish prom=False
2024-04-15 12:55:49,862 readfish toml='canola_test_1600_N11_barcode1-4_bed.toml'
2024-04-15 12:55:49,862 readfish.validate .
.
.
.
2024-04-15 12:55:49,878 readfish.validate Loaded TOML config without error
2024-04-15 12:55:49,878 readfish.validate Initialising Caller
2024-04-15 12:55:49,919 readfish.validate Caller initialised
2024-04-15 12:55:49,920 readfish.validate Initialising Aligner
2024-04-15 12:55:51,695 readfish.validate Aligner initialised
2024-04-15 12:55:51,700 readfish.validate Configuration description:
Number of barcodes in the Conf (excluding unclassified and classified): 4
Barcode unclassified_reads (control=False), Barcode classified_reads (control=False), Barcode NAM00_300csv (control=False), Barcode NAM01_300csv (control=False), Barcode NAM04_300csv (control=False) and Barcode NAM05_300csv (control=False)

Region canola_test (control=False).
Region applies to section of flow cell (# = applied, . = not applied):

    ################################


NOTE - The following 315 contigs are listed as targets but have not been found on the target reference:
 4546	108529	108529	+, 777	510021	510052	+, N11	1	58419410	+, N12	1	5639592	+, N13	1	5639592	+, N19	1	5639592	+, N2	10057805	10064396	+, N2	100
```

But my target in csv got validated


```
`(rf_con) cxo314@gifs-c36qc14:/data/reference/canola$ head adaONT_300SVcsv.txt
777,510021,510052,+
4546,108529,108529,+
N11,1,58419410,+
N12,1,5639592,+
N13,1,5639592,+
N19,1,5639592,+
N2,68644,69237,+
N2,190124,190157,+
N2,211196,213168,+
N2,310498,310546,+
```
```
(rf_con) cxo314@gifs-c36qc14:/data/reference/canola$ readfish validate canola_test_1600_N11_barcode1-4_csv.toml 
2024-04-15 13:10:53,713 readfish /home/cxo314/gifs/.conda/envs/rf_con/bin/readfish validate canola_test_1600_N11_barcode1-4_csv.toml
2024-04-15 13:10:53,714 readfish command='validate'
2024-04-15 13:10:53,714 readfish log_file=None
2024-04-15 13:10:53,714 readfish log_format='%(asctime)s %(name)s %(message)s'
2024-04-15 13:10:53,714 readfish log_level='info'
2024-04-15 13:10:53,714 readfish no_check_plugins=False
2024-04-15 13:10:53,714 readfish no_describe=False
2024-04-15 13:10:53,714 readfish prom=False
2024-04-15 13:10:53,714 readfish toml='canola_test_1600_N11_barcode1-4_csv.toml'
2024-04-15 13:10:53,714 readfish.validate eJztWE1v3DYQvetXEPIlRtfaXccGEgM5OCkSBEjsIHF6MVyBK1Er1hKpiJQ/.
.
.
.
2024-04-15 13:10:53,727 readfish.validate Loaded TOML config without error
2024-04-15 13:10:53,727 readfish.validate Initialising Caller
2024-04-15 13:10:53,770 readfish.validate Caller initialised
2024-04-15 13:10:53,770 readfish.validate Initialising Aligner
2024-04-15 13:10:55,572 readfish.validate Aligner initialised
2024-04-15 13:10:55,573 readfish.validate Configuration description:
Number of barcodes in the Conf (excluding unclassified and classified): 4
Barcode unclassified_reads (control=False), Barcode classified_reads (control=False), Barcode NAM00_300csv (control=False), Barcode NAM01_300csv (control=False), Barcode NAM04_300csv (control=False) and Barcode NAM05_300csv (control=False)

Region canola_test (control=False).
Region applies to section of flow cell (# = applied, . = not applied):

2024-04-15 13:10:57,011 readfish.validate Using the mappy_rs plugin. Using reference: /data/reference/canola/N99_hifi.mmi.

Region canola_test has targets on 8 contigs, with 8 found in the provided reference.
This region has 314 total targets (+ve and -ve strands), covering approximately 3.76% of the genome.

Barcode unclassified_reads has targets on 0 contigs, with 0 found in the provided reference.
This barcode has 0 total targets (+ve and -ve strands), covering approximately 0.00% of the genome.

Barcode classified_reads has targets on 0 contigs, with 0 found in the provided reference.
This barcode has 0 total targets (+ve and -ve strands), covering approximately 0.00% of the genome.

Barcode NAM00_300csv has targets on 8 contigs, with 8 found in the provided reference.
This barcode has 314 total targets (+ve and -ve strands), covering approximately 3.76% of the genome.

Barcode NAM01_300csv has targets on 8 contigs, with 8 found in the provided reference.
This barcode has 314 total targets (+ve and -ve strands), covering approximately 3.76% of the genome.

Barcode NAM04_300csv has targets on 8 contigs, with 8 found in the provided reference.
This barcode has 314 total targets (+ve and -ve strands), covering approximately 3.76% of the genome.

Barcode NAM05_300csv has targets on 8 contigs, with 8 found in the provided reference.
This barcode has 314 total targets (+ve and -ve strands), covering approximately 3.76% of the genome.
`
```

**Comment by mattloose** [comment]
Hi,

The issue is that your BED file is not the correct format. If you look in the docs here - https://looselab.github.io/readfish/toml.html#bed-or-csv-targets

we specify that for a BED file it needs to be in the 6 column format:

chrom chromStart chromEnd name score strand

name and score can be left as a "." if no information is being provided. However, the four column format you provided above does not match the 6 column format.

You would need:

`777	510021	520052	.	.	+`

For the CSV format we do take the four columns. 

Hope that helps.

**Comment by bioteksampath** [comment]
Thanks @mattloose  the 6 cloumn .bed worked. (seems files name should end with .bed, if **bed.txt** not works)
